### ISSUE TITLE
Finding uncaught checked exception

### ISSUE THREAD

**Issue by Mygod** [arturbosch/detekt#721]
https://discuss.kotlinlang.org/t/in-future-could-kotlin-have-checked-exception/1579/8

> I would very much like a static analysis option to show all the possible lines where checked exceptions could be generated and escape (uncaught).

Even though checked exceptions are very bad for functional programming, it's still pretty useful sometimes to have this check to know what exception we forgot to catch. :)

**Comment by Mauin** [comment]
I am guessing this would need type resolution as we would have to check the signature of each function that we call to see if it would throw?

**Comment by Mygod** [comment]
Yes. And then we probably need to handle dependency issues (exception declared in dependencies)... If only JetBrains could do this in their compiler or IDE...

**Comment by schalkms** [comment]
Furthermore, one would also have to check java sources when interop is used.

**Comment by arhont375** [comment]
Just curious is there was any progress on it?
@3flex do you have some idea how to implement it?

**Comment by 3flex** [comment]
No progress, but I was planning to look into it soon.

It should be easy enough to determine if the callee throws an exception that is a subclass of Throwable (but not RuntimeException or Error) but will have to see how that works with Java interop as mentioned above.

If it's challenging I might abandon it for a while as I'm focusing on false positives, bugs and low hanging fruit at the moment.

**Comment by arhont375** [comment]
@3flex Thanks for details!
I have interest in providing such functionality so would like to try to contribute if you will not have time.

**Comment by 3flex** [comment]
That would be great! Please go ahead, the more external contributes we have the better :)

**Comment by schalkms** [comment]
@arhont375 do you have updates on this rule for us? We gladly assist if you have any questions or need help.

**Comment by arhont375** [comment]
Sorry for the delay, so many things happened since then and I got a bit busy so after initial investigation I didn't do much :cry: 
If someone else would be able to take it please go ahead
